xbuild/src/lib.rs

```diff
@@ -316,7 +320,13 @@ impl CompileTarget {
     }
 
     pub fn is_host(self) -> Result<bool> {
-        Ok(self.platform() == Platform::host()? && self.arch() == Arch::host()?)
+        // 32 bit binaries can run on 64 bit hardware, of particular use for testing
+        // This way 32 bit xbuild will produce 32 bit binaries, unless commanded otherwise
+        if cfg!(target_arch = "x86") {
+            Ok(false)
```

Comment:
Not sure I follow. This doesn't seem correct as you might be breaking fully-native builds here? That's where `xbuild` should return `true` for this function.